dimod/responses/numpy_response.py

```diff
@@ -290,11 +290,23 @@ def add_samples_from_array(self, samples, energies, sample_data=None, sorted_by_
 
     def samples_array(self):
         """Returns the :obj:`numpy.ndarray` containing the samples."""
-        return self._samples
+        import numpy as np
```

Comment:
A few general questions:
1. should samples/energy_array be a `@property`? 
2. why store self._samples in a samples name and not just return self._samples?

A nitpick:
`return np.empty((0, 0), dtype=int) if self._samples is None else self.samples`?